### PR TITLE
drivers: i2c: it8xxx2: fix i2c_reset issue

### PR DESCRIPTION
**Pull request by GTLin08**
After i2c_reset, there is still no external pull-up I2C bus,
and finishing off the rest of loop causes the code to hang
indefinitely.
This patch fixes that if I2C bus is not available(No external
pull-up), dropping the transaction.

Another issue is fixed the slave address from 7-bit to 8-bit.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

### CODE REVIEW

File: drivers/i2c/i2c_ite_it8xxx2.c [hunk 1]
```diff
@@ -721,6 +732,18 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		LOG_ERR("Device message is NULL");
 		return -EINVAL;
 	}
+	/* Make sure we're in a good state to start */
+	if (i2c_bus_not_available(dev)) {
+		/* reset i2c port */
+		i2c_reset(dev, I2C_RC_NO_IDLE_FOR_START);
+		/*
+		 * After resetting I2C bus, if I2C bus is not available
+		 * (No external pull-up), dropping the transaction.
```
Comment (sjg20) [inline drivers/i2c/i2c_ite_it8xxx2.c:741]: s/dropping/drop/

File: drivers/i2c/i2c_ite_it8xxx2.c [hunk 2]
```diff
@@ -730,14 +753,8 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		data->ridx = 0;
 		data->err = 0;
 		data->msgs = &(msgs[i]);
-		data->addr_16bit = addr;
+		data->addr_16bit = addr << 1;
```
Comment (brockus-zephyr) [inline drivers/i2c/i2c_ite_it8xxx2.c:756]: This change is not related to the i2c_reset issue. Can it be broken out into a PR for fixing the i2c 7 bit address.

File: drivers/i2c/i2c_ite_it8xxx2.c [hunk 3]
```diff
@@ -172,6 +172,16 @@ static int i2c_is_busy(const struct device *dev)
 	return (IT83XX_I2C_STR(base) & E_HOSTA_BB);
 }
 
+static int i2c_bus_not_available(const struct device *dev)
+{
+	if ((i2c_is_busy(dev)) ||
```
Comment (sjg20) [inline drivers/i2c/i2c_ite_it8xxx2.c:177]: drop at least these extra brackets

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 2 above)
Comment (sjg20) [inline drivers/i2c/i2c_ite_it8xxx2.c:756]: Can you not use a 7-bit address in the driver? It seems a bit confusion to change it

Comment (keith-zephyr) [inline drivers/i2c/i2c_ite_it8xxx2.c:756]: The TRASLA register already splits the I2C target address into ADR [7:1] and R/W [0] fields.
Perform this shift when writing to the TRASLA register to better match the datasheet.

IT83XX_SMB_TRASLA(base) = (uint8_t)(data->addr_16bit << 1 ) | 0x01;

Do the same thing when calling i2c_pio_trans_data.

**Review by sjg20 — CHANGES_REQUESTED** [review]

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 3 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c]: Done

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 1 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c]: Done

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 2 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c]: OK, it has be split another commit:
https://github.com/zephyrproject-rtos/zephyr/pull/34809/commits/c150ec4f73641e522e2ac18a1f9da4eb5dff1f9c

Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c]: OK, removed.

File: drivers/i2c/i2c_ite_it8xxx2.c [hunk 4]
```diff
@@ -746,7 +763,13 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		/* Start transaction */
 		i2c_transaction(dev);
 		/* Wait for the transfer to complete */
-		k_sem_take(&data->device_sync_sem, K_FOREVER);
+		/* TODO: the timeout should be adjustable */
+		res = k_sem_take(&data->device_sync_sem, K_MSEC(100));
+		if (res != 0) {
+			data->err = ETIMEDOUT;
+			/* reset i2c port */
+			i2c_reset(dev, I2C_RC_TIMEOUT);
```
Comment (sjg20) [inline drivers/i2c/i2c_ite_it8xxx2.c:771]: Why continue with the next message if this one failed?

**Review by sjg20 — CHANGES_REQUESTED** [review]

**Review by keith-zephyr — APPROVED** [review]

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 2 above)
Comment (keith-zephyr) [reply drivers/i2c/i2c_ite_it8xxx2.c]: That appears to be a bad link.

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 4 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c:771]: Yes, this transaction should be dropped. It's fixed.

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 2 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c]: What about this link? https://github.com/zephyrproject-rtos/zephyr/pull/34809/commits/8f70f58c90eeb9ea303f07a6f30adfd862803337

File: drivers/i2c/i2c_ite_it8xxx2.c [hunk 5]
```diff
@@ -746,7 +763,15 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
...
+			/* If this message is sent fail, drop the transaction. */
+			return -EINVAL;
```
Comment (sjg20) [inline drivers/i2c/i2c_ite_it8xxx2.c:773]: Why -EINVAL? Should it not be -ETIMEDOUT ?

Why is there a separate data->err value? Is that because the errors happen in interrupts?

**Review by sjg20 — CHANGES_REQUESTED** [review]

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 5 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c]: There should be modified to "break;".  If k_sem_take is time out, it will break and then run the follow-up program and return ETIMEDOUT. 

data->err stores the error status of the transaction and it will be returned in this function.

**Comment by Dino-Li** [comment]
Any comments will be appreciated.

**Review by sjg20 — APPROVED** [review]

**Review by brockus-zephyr — APPROVED** [review]

**Comment by Dino-Li** [comment]
Can we invite authorized user to join this PR?
